llama.cpp

```diff
@@ -3703,6 +3729,21 @@ static bool llama_model_load(const std::string & fname, llama_model & model, con
             return true;
         }
 
+#ifdef GGML_USE_KOMPUTE
+        if (ggml_vk_has_device() && params.n_gpu_layers > 0 && (
+            !(model.arch == LLM_ARCH_LLAMA || model.arch == LLM_ARCH_FALCON)
+            || !(
+                model.ftype == LLAMA_FTYPE_ALL_F32 ||
+                model.ftype == LLAMA_FTYPE_MOSTLY_F16 ||
+                model.ftype == LLAMA_FTYPE_MOSTLY_Q4_0 ||
+                model.ftype == LLAMA_FTYPE_MOSTLY_Q4_1
+            )
+        )) {
+            // disable Vulkan due to unsupported model architecture or quantization type
+            params.n_gpu_layers = 0;
+        }
+#endif
```

Comment:
@ggerganov This is a hack that I think will have to change before this can be merged. We're doing it this way because we need to know the arch and ftype of the model in order to know if we can load it, otherwise loading will fail and the user won't know why. This function only returns true or false, but we need to tell the difference between "out of VRAM" and "unsupported quantization type".

Reply:
From `llama.cpp` perspective, it is fine as it is. We just have to add a warning message and it should be OK. Returning false would be better, but AFAIU it will be more difficult for 3rd party projects to handle this.